server/src/catalog.rs

```diff
@@ -216,72 +311,248 @@ fn read_model_table_metadata(
-    rows: &RecordBatch,
-    column_index: usize,
-) -> Result<ArrayRef, ParquetError> {
-    let schema = rows.schema();
-    let name = schema.field(column_index).name();
-    let column = rows
-        .column(column_index)
-        .as_any()
-        .downcast_ref::<BinaryArray>()
-        .unwrap();
-    let mut shifted_column =
-        StringBuilder::with_capacity(column.len(), column.get_buffer_memory_size());
-    shifted_column.append_value(name)?; //The level's name is stored at index 0 as ids starting at 1 is used for lookup
-    for member_as_bytes in column {
-        let member_as_str = str::from_utf8(member_as_bytes.unwrap())?;
-        shifted_column.append_value(member_as_str)?;
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datafusion::arrow::array::StringArray;
+    use proptest::num;
+    use proptest::string;
+    use proptest::{prop_assert, prop_assert_eq, proptest};
+
+    // Tests for Catalog.
+    #[test]
+    fn test_empty_catalog_table_and_model_table_names() {
+        let catalog = Catalog {
+            table_metadata: vec![],
+            model_table_metadata: vec![],
+        };
+        assert!(catalog.table_and_model_table_names().is_empty())
+    }
+
+    #[test]
+    fn test_catalog_table_and_model_table_names() {
+        let mut catalog = Catalog {
+            table_metadata: vec![],
+            model_table_metadata: vec![],
+        };
+
+        catalog.table_metadata.push(TableMetadata {
+            name: "table".to_owned(),
+            path: "".to_owned(),
+        });
+
+        catalog
+            .model_table_metadata
+            .push(Arc::new(ModelTableMetadata {
+                name: "model_table".to_owned(),
+                segment_path: ObjectStorePath::parse("").unwrap(),
+                segment_file_legacy_schema: Arc::new(Schema::new(vec![])),
+                sampling_intervals: Int32Array::builder(0).finish(),
+                denormalized_dimensions: vec![],
+            }));
+
+        assert_eq!(
+            vec!("table", "model_table"),
+            catalog.table_and_model_table_names()
+        )
+    }
+
+    // Tests for TableMetadata.
+    #[test]
+    fn test_table_metadata_zero_suffixes() {
+        let table_metadata = TableMetadata::new("folder".to_owned(), "/path/to/folder".to_owned());
+        assert_eq!("folder", table_metadata.name);
+    }
+
+    #[test]
+    fn test_table_metadata_single_suffix() {
+        let table_metadata =
+            TableMetadata::new("file.parquet".to_owned(), "/path/to/folder".to_owned());
+        assert_eq!("file", table_metadata.name);
+    }
+
+    #[test]
+    fn test_table_metadata_multiple_suffixes() {
+        let table_metadata = TableMetadata::new(
+            "file.snappy.parquet".to_owned(),
+            "/path/to/folder".to_owned(),
+        );
+        assert_eq!("file", table_metadata.name);
+    }
+
+    // Tests for ModelTableMetadata.
+    #[test]
+    fn test_extract_and_shift_empty_int32_array() {
```

Comment:
Very minor but maybe add some newlines to this test, `test_extract_and_shift_int32_array_with_value`, `test_extract_and_shift_empty_text_array`, and `test_extract_and_shift_text_array_with_value`. They are a bit hard to read currently.

Reply:
Fixed.